### PR TITLE
Add third person controller 

### PR DESCRIPTION
**Pull request by Jade-JadeH**
**Summary**

The third person controller is based on [Starter Assets - Third Person Character Controller ](https://assetstore.unity.com/packages/essentials/starter-assets-third-person-character-controller-196526). 

1. Modified `ThirdPersonController.cs` to add a wait jump until collider generated function (**line 163-line 179** is what I modified). 
2. Add `ChangeMaterial.cs` to swap between HDRP and URP mapman materials when the pipeline changes
3. Modify `LightingManager.cs` to swap URP skybox. I do notice that the skybox doesn't follow natural science in the stream layer sample (because when you see from the outer space you won't see the earth atmosphere) but this is a minor issue
4. Resolved a small issue on the measurement sample

**ArcGIS Maps SDK Version**

Tested with daily 3717

### CODE REVIEW

**Comment by mrnelson6** [comment]
There are a lot of changes here. Can you add a description on this PR and just cover what it is all changing since you are doing more than just adding the third person sample.

Also a better title should be used

File: samples_project/Assets/StarterAssets/Editor/StarterAssetsDeployMenu.cs [hunk 1]
```diff
@@ -0,0 +1,169 @@
+using System;
+using System.Linq;
+using StarterAssetsPackageChecker;
```
Comment (mrnelson6) [inline samples_project/Assets/StarterAssets/Editor/StarterAssetsDeployMenu.cs:3]: This line was causing an error in my script when I loaded it. 

This starterasset is quite heavy and it is bringing a lot into our repo and adding a lot of dependencies

File: samples_project/Assets/StarterAssets/Editor/PackageChecker/StarterAssetsPackageChecker.dll.meta [hunk 2]
```diff
@@ -0,0 +1,33 @@
+fileFormatVersion: 2
```
Comment (mrnelson6) [inline samples_project/Assets/StarterAssets/Editor/PackageChecker/StarterAssetsPackageChecker.dll.meta:1]: this .dll.meta is comitted but there is no .dll committed.

**Review by mrnelson6 — DISMISSED** [review]
Your PR is also missing a project dependency on Cinemachine which is required. 

In short, I think we can accomplish what this PR is trying to show without bringing all of this in.

**Review by NaumanIsGISMan — CHANGES_REQUESTED** [review]
I feel this is a massive amount of cruft to be bringing into the repo. I dont think we should be doing this, is there a way to add the package separately or something? I know its nice to have the robot dude as itll match Unreal, but I would think simpler would be better in this case

File: samples_project/Assets/StarterAssets/Editor/PackageChecker/StarterAssetsPackageChecker.dll.meta (hunk 2 above)
Comment (Jade-JadeH) [reply samples_project/Assets/StarterAssets/Editor/PackageChecker/StarterAssetsPackageChecker.dll.meta:1]: > Your PR is also missing a project dependency on Cinemachine which is required.
> 
> In short, I think we can accomplish what this PR is trying to show without bringing all of this in.

so you mean to still use this starter asset but select those scripts that are used?

**Comment by Jade-JadeH** [comment]
> I feel this is a massive amount of cruft to be bringing into the repo. I dont think we should be doing this, is there a way to add the package separately or something? I know its nice to have the robot dude as itll match Unreal, but I would think simpler would be better in this case

yeah I could completely remove the default mannequin. I use mapman anyways.

File: samples_project/Assets/ChangeMaterial.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Rendering;
+
+public class ChangeMaterial : MonoBehaviour
+{
+    // Start is called before the first frame update
+    void Start()
+    {
+        var pipeline = GraphicsSettings.defaultRenderPipeline;
```
Comment (NaumanIsGISMan) [inline samples_project/Assets/ChangeMaterial.cs:11]: I dont see where this is being used
```suggestion
```

File: samples_project/Assets/ChangeMaterial.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        SkinnedMeshRenderer mapMan = GetComponent<SkinnedMeshRenderer>();
+        Material mapManMat = null;
+
+        if (GraphicsSettings.renderPipelineAsset.name.Contains("HDRP"))
+        {
+            mapManMat = Resources.Load("MapmanForUnity/HDRP_Metallic_Standard/MapmanMaterialHDRP", typeof(Material)) as Material;
+        }
+        else if (GraphicsSettings.renderPipelineAsset.name.Contains("URP"))
```
Comment (NaumanIsGISMan) [inline samples_project/Assets/ChangeMaterial.cs:19]: `GraphicsSettings.renderPipelineAsset` is deprecated, would maybe instead use [GraphicsSettings.defaultRenderPipeline](https://docs.unity3d.com/ScriptReference/Rendering.GraphicsSettings-defaultRenderPipeline.html)

https://docs.unity3d.com/ScriptReference/Rendering.GraphicsSettings-renderPipelineAsset.html

**Review by NaumanIsGISMan — CHANGES_REQUESTED** [review]
Working on reviewing this. This is part one

**Comment by bwbair01** [comment]
Getting the following error when trying to test the project...

Problem detected while importing the Prefab file: 'Assets/StarterAssets/ThirdPersonController/Prefabs/NestedParentCapsule_Unpack.prefab'.
The file might be corrupt or have missing nested Prefabs. See details below.
Errors:
	Prefab instance problem: UI_EventSystem (Missing Prefab with guid: f0271df749728104eac22c3d897fd8ce)
	Prefab instance problem: UI_Canvas_StarterAssetsInputs_Joysticks (Missing Prefab with guid: 2f7f3dde7ae722a4aafffe20691ad702)

I'm using unity 2021.3.2f1 and SDK 1.1.0 release

**Comment by Jade-JadeH** [comment]
> Getting the following error when trying to test the project...
> 
> Problem detected while importing the Prefab file: 'Assets/StarterAssets/ThirdPersonController/Prefabs/NestedParentCapsule_Unpack.prefab'. The file might be corrupt or have missing nested Prefabs. See details below. Errors: Prefab instance problem: UI_EventSystem (Missing Prefab with guid: f0271df749728104eac22c3d897fd8ce) Prefab instance problem: UI_Canvas_StarterAssetsInputs_Joysticks (Missing Prefab with guid: 2f7f3dde7ae722a4aafffe20691ad702)
> 
> I'm using unity 2021.3.2f1 and SDK 1.1.0 release

I just fresh cloned the repo and tested and didn't encounter the error. If you checkout my branch directly from your branch maybe there're some conflicts. Could you test with a clean repo?

**Comment by bwbair01** [comment]
I did use a clean copy of the repo, not my branch.  Could it be due to using the 1.1.0 release instead of the daily build maps SDK?

**Comment by Jade-JadeH** [comment]
> I did use a clean copy of the repo, not my branch. Could it be due to using the 1.1.0 release instead of the daily build maps SDK?

I switched to a different computer but still couldn't reproduce... I used 2021.3+maps sdk released version

**Comment by bwbair01** [comment]
> 

Ok, I tried that and I am able to run the sample. Still looking over all the changes, will post any questions/comments soon.

File: samples_project/Packages/manifest.json [hunk 5]
```diff
@@ -1,6 +1,8 @@
 {
   "dependencies": {
+    "com.esri.arcgis-maps-sdk": "file:C:/software/com.esri.arcgis-maps-sdk-1.1.0_3717.tgz",
     "com.unity.burst": "1.6.5",
+    "com.unity.cinemachine": "2.8.9",
```
Comment (bwbair01) [inline samples_project/Packages/manifest.json:5]: I am not against pulling in cinemachine, as long as @NaumanIsGISMan is ok with it... 

I understand Matt's previous comment about making a simplified controller with fewer dependencies; however, given the complexities and corner-cases of map/building geometry I do think that using Unity's robust ThirdPersonController and cinemachine results in a better UX in this case.

File: samples_project/ProjectSettings/ProjectVersion.txt [hunk 6]
```diff
@@ -1,2 +1,2 @@
-m_EditorVersion: 2021.3.2f1
-m_EditorVersionWithRevision: 2021.3.2f1 (d6360bedb9a0)
+m_EditorVersion: 2021.3.1f1
```
Comment (bwbair01) [inline samples_project/ProjectSettings/ProjectVersion.txt:1]: Should revert this file and/or install 2021.3.2f1 locally

**Review by bwbair01 — CHANGES_REQUESTED** [review]
I made some suggestions; I don't think I missed anything.

One question, though, should we really be adding all the sound effects for character movement? I don't think they're necessary for demonstrating a third person controller, and none of the other samples use sound.

File: samples_project/Assets/SampleViewer/SampleViewer.unity [hunk 7]
```diff
@@ -38,7 +38,7 @@ RenderSettings:
   m_ReflectionIntensity: 1
   m_CustomReflection: {fileID: 0}
   m_Sun: {fileID: 0}
-  m_IndirectSpecularColor: {r: 0, g: 0, b: 0, a: 1}
+  m_IndirectSpecularColor: {r: 262.3369, g: 325.0498, b: 430.21924, a: 1}
```
Comment (ZackAllen) [inline samples_project/Assets/SampleViewer/SampleViewer.unity:41]: Is this change intended?

File: samples_project/Assets/SampleViewer/Samples/Measure/Measure.unity [hunk 8]
```diff
@@ -24,7 +24,7 @@ RenderSettings:
   m_AmbientEquatorColor: {r: 0.114, g: 0.125, b: 0.133, a: 1}
   m_AmbientGroundColor: {r: 0.047, g: 0.043, b: 0.035, a: 1}
   m_AmbientIntensity: 1
-  m_AmbientMode: 0
+  m_AmbientMode: 4
```
Comment (ZackAllen) [inline samples_project/Assets/SampleViewer/Samples/Measure/Measure.unity:27]: I don't think Measure changes should go in the third person controller PR. Better to keep these changes atomic.

File: samples_project/Assets/SampleViewer/Samples/ThirdPerson/ThirdPerson.unity [hunk 9]
```diff
@@ -0,0 +1,3506 @@
...
+  m_Modification:
+    m_TransformParent: {fileID: 8002264681162914670}
+    m_Modifications:
+    - target: {fileID: -8679921383154817045, guid: 6392c7aea3c8f634e9bd5139bb16d63c, type: 3}
+      propertyPath: m_RootOrder
+      value: 0
+      objectReference: {fileID: 0}
+    - target: {fileID: -8679921383154817045, guid: 6392c7aea3c8f634e9bd5139bb16d63c, type: 3}
+      propertyPath: m_LocalPosition.x
+      value: 0.36
```
Comment (ZackAllen) [inline samples_project/Assets/SampleViewer/Samples/ThirdPerson/ThirdPerson.unity:866]:
```suggestion
      value: 0
```
This makes the player rotation awkward, works better at 0.

File: samples_project/Assets/SampleViewer/Samples/ThirdPerson/ThirdPerson.unity [hunk 10]
```diff
@@ -0,0 +1,3506 @@
...
+      objectReference: {fileID: 0}
+    - target: {fileID: -8679921383154817045, guid: 6392c7aea3c8f634e9bd5139bb16d63c, type: 3}
+      propertyPath: m_LocalPosition.y
+      value: 0
+      objectReference: {fileID: 0}
+    - target: {fileID: -8679921383154817045, guid: 6392c7aea3c8f634e9bd5139bb16d63c, type: 3}
+      propertyPath: m_LocalPosition.z
+      value: 0.17
```
Comment (ZackAllen) [inline samples_project/Assets/SampleViewer/Samples/ThirdPerson/ThirdPerson.unity:874]:
```suggestion
      value: 0
```
This makes the player rotation awkward, works better at 0.

File: samples_project/Assets/SampleViewer/Samples/ThirdPerson/readme.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
+# Explore with a Third Person Character
+
+Allows users to explore from the perspective of a Third Person Character.
```
Comment (ZackAllen) [inline samples_project/Assets/SampleViewer/Samples/ThirdPerson/readme.md:3]:
```suggestion
Explore from the perspective of a third person camera following a controllable character.
```
We shouldn't us user story type language here. The users are already consuming this as public doc.

File: samples_project/Assets/SampleViewer/Samples/ThirdPerson/readme.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
+# Explore with a Third Person Character
```
Comment (ZackAllen) [inline samples_project/Assets/SampleViewer/Samples/ThirdPerson/readme.md:1]:
```suggestion
# Explore with a Third Person Character Controller
```
The character isn't third person, the camera/controller is. Not sure about this title, I'm open to new ideas.

File: samples_project/Assets/SampleViewer/Samples/ThirdPerson/readme.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+## About the data
+
+Building models for San Francisco are loaded from a [3D object scene layer](https://tiles.arcgis.com/tiles/z2tnIkrLQ2BRzr6P/arcgis/rest/services/SanFrancisco_Bldgs/SceneServer) hosted by Esri.
+
+Elevation data is loaded from the [Terrain 3D elevation layer](https://www.arcgis.com/home/item.html?id=7029fb60158543ad845c7e1527af11e4) hosted by Esri.
+
+## Tags
+
+exploration, third person perspective, third person controller
```
Comment (ZackAllen) [inline samples_project/Assets/SampleViewer/Samples/ThirdPerson/readme.md:33]:
```suggestion
3rd person, camera, controller, exploration, follow, platformer, third person perspective, third person controller
```

File: samples_project/Assets/SampleViewer/Samples/ThirdPerson/readme.metadata.json [hunk 14]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (ZackAllen) [inline samples_project/Assets/SampleViewer/Samples/ThirdPerson/readme.metadata.json:1]: This should be updated to match the readme when the readme changes are final.

**Review by ZackAllen — CHANGES_REQUESTED** [review]

File: samples_project/Assets/SampleViewer/Samples/ThirdPerson/readme.md (hunk 11 above)
Comment (Jade-JadeH) [reply samples_project/Assets/SampleViewer/Samples/ThirdPerson/readme.md:3]: That makes sense about the tone, I just completely removed this sentence since the title already tells it all

File: samples_project/Assets/SampleViewer/Samples/Measure/Measure.unity (hunk 8 above)
Comment (ZackAllen) [reply samples_project/Assets/SampleViewer/Samples/Measure/Measure.unity:27]: Can we fix it in a separate PR? Bad practice to include in an unrelated PR.

Comment (Jade-JadeH) [reply samples_project/Assets/SampleViewer/Samples/Measure/Measure.unity:27]: ok, yeah there're some other things that I can improve in the measure PR too so I can do them altogether

File: samples_project/Packages/manifest.json (hunk 5 above)
Comment (NaumanIsGISMan) [reply samples_project/Packages/manifest.json:4]: Yeah, I think at this point I'm cool it.

**Review by NaumanIsGISMan — COMMENTED** [review]
Just be sure when you get the chance to revert those changes to the measurement sample

File: samples_project/Assets/SampleViewer/Samples/Measure/Measure.unity (hunk 8 above)
Comment (Jade-JadeH) [inline samples_project/Assets/SampleViewer/Samples/Measure/Measure.unity:27]:
```suggestion
  m_AmbientMode: 0
```

**Review by NaumanIsGISMan — APPROVED** [review]
